visualization/app/codeCharta/ui/ribbonBar/ribbonBar.component.ts

```diff
@@ -1,49 +1,26 @@
 import "./ribbonBar.component.scss"
 import $ from "jquery"
-import { IRootScopeService } from "angular"
-import { FileState } from "../../codeCharta.model"
-import { FileStateService, FileStateServiceSubscriber } from "../../state/fileState.service"
-import { FileStateHelper } from "../../util/fileStateHelper"
 
-export class RibbonBarController implements FileStateServiceSubscriber {
+export class RibbonBarController {
 	private collapsingElements = $(
 		"code-map-component #codeMap, ribbon-bar-component #header, ribbon-bar-component .section-body, #toggle-ribbon-bar-fab"
 	)
 	private toggleElements = $("ribbon-bar-component .section-title")
 	private isExpanded: boolean = false
 
-	private _viewModel: {
-		isDeltaState: boolean
-	} = {
-		isDeltaState: null
-	}
-
-	/* @ngInject */
-	constructor(private $rootScope: IRootScopeService) {
-		FileStateService.subscribe(this.$rootScope, this)
-	}
-
-	public onFileSelectionStatesChanged(fileStates: FileState[], event: angular.IAngularEvent) {
-		this._viewModel.isDeltaState = FileStateHelper.isDeltaState(fileStates)
-	}
-
-	public onImportedFilesChanged(fileStates: FileState[], event: angular.IAngularEvent) {}
-
 	public toggle() {
-		if (!this.isExpanded) {
-			this.expand()
-		} else {
+		if (this.isExpanded) {
 			this.collapse()
+		} else {
+			this.expand()
 		}
 	}
 
 	public expand() {
-		this.isExpanded = true
```

Comment:
Why did you remove these lines? The ribbon bar can't be collapsed again without them.

Reply:
And why are there no tests that ensure that this behaviour works?
isExpanded should be in the _viewModel and you can easily unit test this

Reply:
Oh yea that was unintended. But its not gonna be inside a _viewModel, because it's not used inside the html template

Reply:
And I added some e2e tests